scalanet/ut/src/io/iohk/scalanet/peergroup/DynamicTLSPeerGroupSpec.scala

```diff
@@ -349,6 +375,142 @@ class DynamicTLSPeerGroupSpec extends AsyncFlatSpec with BeforeAndAfterAll {
         }
     }
   }
+
+  it should "not send another message when receive queue and both buffers are full " in backPressureTestCase(
+    clientConfig = getCorrectConfig(maxFrameLength = 64 * 1024 * 1024),
+    serverConfig = getCorrectConfig(maxQueueSize = 1, maxFrameLength = 64 * 1024 * 1024)
+  ) {
+    case (clientChannel, serverChannel) =>
+      // we are using pretty large messages to be sure incoming and sending buffers will be full
```

Comment:
What's the sending buffer size limit?

Reply:
It is really os/tcp implementation dependant, but in case of tests which fills buffers we are using messages large enough to fill them in all cases.

Those config could be exposed externally (as I described in pr description) by:

> Currently we are using default adaptive strategies form RCV and SND buffers sizes. We may consider adding options to size them by the user. Interesting options are : SO_SNDBUF, SO_RCVBUF and WRITE_BUFFER_WATER_MARK. They decide when change channel writeablitiy status. We can do follow up task to expose them to the user.

But even then options `SO_SNDBUF` and `SO_RCVBUF` are only hints to OS, and in some cases OS doubles this values, so we can't relay on them exactly in tests

Reply:
Right, thanks, I forgot about the PR description.